Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/subdisc/HL7QueryParamsTransforms.java

```diff
@@ -53,12 +53,14 @@
  * @author Jon Hoppesch
  */
 public class HL7QueryParamsTransforms {
```

Comment:
Just a suggestion, HL7MessageIdGenerator.generateHL7MessageId() used in multiple transformer class, we can create a parent class with a wrapper method and use it in the sub class. Also change the HL7MessageIdGenerator to a singleton class. We can address this in future re-factoring effort.